src/shared/modals/DangerZone/DangerZoneAction.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { useState } from 'react';
+import { Button, Col, Form, Input, Modal, Row } from 'antd';
+
+export type TDangerZoneActionProps = {
+  open: boolean;
+  onClose(): void;
+  matchTerm: string;
+  handler(values: any): void;
+  action?: 'deprecate' | 'delete';
```

Comment:
Hmm, in which case can `action` be undefined? It seems weird to me that some component would call this component if there is no `action`.

Reply:
for this if you see in the state that manage the actions, it must be empty (`undefined`) at the beginning if not the action will take either `deprecate` or `delete` and then action selecting the action will not be correct, 
the action property is not only for the text that it's showed but also for the action of the button, 
may be you are right if we have to separate the text from the `action`